#! /usr/bin/env python

PACKAGE='tuw_local_planner'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

mode_enum = gen.enum([ gen.const("Stop",           int_t, 0, "Planner stopped"),
                       gen.const("Demo",           int_t, 1, "Planner in demo mode"),
                       gen.const("Wanderer",       int_t, 2, "Planner in wanderer mode")],
                       "An enum select an operation mode")
gen.add("plot_data",   bool_t,   0, "",  False)
gen.add("mode", int_t, 0, "A size parameter which is edited via an enum", 0, 0, 7, edit_method=mode_enum)
gen.add("map_pix_x", int_t, 0, "map size in pixel", 500, 10, 2000)
gen.add("map_pix_y", int_t, 0, "map size in pixel", 500, 10, 2000)
gen.add("map_max_x", double_t, 0, "map size in meter", 8, -15, 15)
gen.add("map_max_y", double_t, 0, "map size in meter", 8, -15, 15)
gen.add("map_min_x", double_t, 0, "map size in meter",-8, -15, 15)
gen.add("map_min_y", double_t, 0, "map size in meter",-8, -15, 15)
gen.add("map_grid_x", double_t, 0, "grid size in meter", 1, 0, 10)
gen.add("map_grid_y", double_t, 0, "grid size in meter", 1, 0, 10)
gen.add("map_rotation", double_t, 0, "map rotation in radians", 0, -3.14, 3.14)
gen.add("safety_distance", double_t, 0, "safety distance of the robot to obstacles in front", 0.5, 0, 4)

exit(gen.generate(PACKAGE, "tuw_local_planner", "LocalPlanner"))
